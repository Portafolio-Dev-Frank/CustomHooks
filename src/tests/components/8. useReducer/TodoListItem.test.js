import { shallow } from "enzyme"
import { TodoListItem } from "../../../components/8. useReducer/TodoListItem";
import { demosTodos } from "../../fixtures/demoTodos";



describe('pruebas en <TodoListItem></TodoListItem>', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(
        <TodoListItem
            todo={demosTodos[0]}
            index={0}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
        />)

    test('debe renderizarse el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de llamar la funcion de handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalledWith(1);
    });

    test('debe de llamar la funcion handleToggle', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(1);
    });

    test('debe de mostrar el texto correctamente ', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(`1. ${demosTodos[0].desc}`)
    })

    test('debe de tener la clase compelte si el todo.done = true', () => {
        const todo = demosTodos[0];
        todo.done = true;

        const wrapper = shallow(<TodoListItem
        todo= {todo}
        ></TodoListItem>)
        expect(wrapper.find('p').hasClass('complete')).toBe(true)
    })
    


})
