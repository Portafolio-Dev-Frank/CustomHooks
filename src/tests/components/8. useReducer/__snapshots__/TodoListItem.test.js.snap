// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoListItem></TodoListItem> debe renderizarse el componente  1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    APRENDER REACT
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    borrar
  </button>
</li>
`;
